src/components/WalletAuthModal.tsx

```diff
@@ -63,7 +63,9 @@ export const WalletAuthModal = ({ open }: { open: boolean }) => {
     if (ethereum) {
       // convert hex to decimal
       const chainId = parseInt(
-        await ethereum.request({ method: 'eth_chainId' }),
+        ethereum.isMetaMask
+          ? await ethereum.request({ method: 'eth_chainId' })
+          : ethereum.sendAsync({ method: 'eth_chainId' }),
```

Comment:
This works specifically for our TestProvider and is wrapped in the ethers `Web3Provider` class so it's supports virtually all rpc requests